lib/src/protobuf/generated_message.dart

```diff
@@ -50,7 +50,18 @@ abstract class GeneratedMessage {
 
   UnknownFieldSet get unknownFields => _fieldSet._ensureUnknownFields();
 
-  bool get _isReadOnly => false;
+  bool _isFrozen = false;
+  bool get _isReadOnly => _isFrozen;
```

Comment:
Yes that's the simple way, but it only freezes the current object (rather than the tree rooted in this object).